gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemNewIntegrationTest.java

```diff
@@ -677,6 +678,83 @@ public void delete_directory_parallel() throws Exception {
+    GoogleCloudStorageFileSystem gcsFs = newGcsFs(gcsFsOptions, gcsRequestsTracker);
+
+    String bucketName = gcsfsIHelper.sharedBucketName1;
+    URI bucketUri = new URI("gs://" + bucketName + "/");
+    String dirObject = getTestResource();
+
+    gcsfsIHelper.createObjects(bucketName, dirObject + "/d1/d2");
+
+    gcsFs.delete(bucketUri.resolve(dirObject + "/d1/"), /* recursive= */ true);
+
+    assertThat(gcsRequestsTracker.getAllRequestStrings())
+        .containsExactly(
+            listRequestWithTrailingDelimiter(
+                bucketName, dirObject + "/d1/", /* maxResults= */ 1, /* pageToken= */ null),
+            listRequestString(
+                bucketName,
+                /* flatList= */ true,
+                /* includeTrailingDelimiter= */ null,
+                dirObject + "/d1/",
+                "bucket,name,generation",
+                /* maxResults= */ 1024,
+                /* pageToken= */ null),
+            deleteRequestString(bucketName, dirObject + "/d1/d2", /* generationId= */ 1));
+  }
+
+  @Test
+  public void delete_inferred_directory_with_parent() throws Exception {
+    GoogleCloudStorageFileSystemOptions gcsFsOptions =
+        newGcsFsOptions().setStatusParallelEnabled(false).build();
+
+    TrackingHttpRequestInitializer gcsRequestsTracker =
+        new TrackingHttpRequestInitializer(httpRequestsInitializer);
+    GoogleCloudStorageFileSystem gcsFs = newGcsFs(gcsFsOptions, gcsRequestsTracker);
+
+    String bucketName = gcsfsIHelper.sharedBucketName1;
+    URI bucketUri = new URI("gs://" + bucketName + "/");
+    String dirObject = getTestResource();
+
+    gcsfsIHelper.createObjects(bucketName, dirObject + "/d1/d2/d3");
+
+    gcsFs.delete(bucketUri.resolve(dirObject + "/d1/d2"), /* recursive= */ true);
+
+    assertThat(gcsFs.exists(bucketUri.resolve(dirObject + "/d1/"))).isTrue();
+
+    assertThat(gcsRequestsTracker.getAllRequestStrings())
+        .containsExactly(
+            getRequestString(bucketName, dirObject + "/d1/d2"),
+            getRequestString(bucketName, dirObject + "/d1/"),
+            uploadRequestString(bucketName, dirObject + "/d1/", /* generationId= */ null),
+            listRequestWithTrailingDelimiter(
+                bucketName, dirObject + "/d1/d2/", /* maxResults= */ 1, /* pageToken= */ null),
+            listRequestWithTrailingDelimiter(
+                bucketName, dirObject + "/d1/", /* maxResults= */ 1, /* pageToken= */ null),
+            listRequestString(
+                bucketName,
+                /* flatList= */ true,
+                /* includeTrailingDelimiter= */ null,
+                dirObject + "/d1/d2/",
+                "bucket,name,generation",
+                /* maxResults= */ 1024,
+                /* pageToken= */ null),
+            deleteRequestString(bucketName, dirObject + "/d1/d2/d3", /* generationId= */ 1));
+  }
+
   private void delete_directory(boolean parallelStatus) throws Exception {
```

Comment:
I think this internal test method was intended to follow public test methods that use it.

Reply:
Yup, makes sense. moved the method back